File: 2026/09-September/17-Thursday/solution.cs
using System;
using System.Collections.Generic;

public class Solution
{
    public int MinSumOfLengths(int[] arr, int target)
    {
        int n = arr.Length;
        int INF = 1_000_000_000;

        // prefix sum -> latest index
        Dictionary<int, int> prefix = new Dictionary<int, int>();
        prefix[0] = 0;

        // best[i] = minimum length of a valid subarray
        // completely inside the first i elements
        int[] best = new int[n + 1];

        for (int i = 0; i <= n; i++)
            best[i] = INF;

        int sum = 0;
        int ans = INF;

        for (int i = 1; i <= n; i++)
        {
            sum += arr[i - 1];

            // Carry forward the previous best
            best[i] = best[i - 1];

            // Look for a previous prefix sum
            if (prefix.ContainsKey(sum - target))
            {
                int j = prefix[sum - target];

                // Subarray [j, i - 1]
                int length = i - j;

                // best[j] is completely before this subarray
                if (best[j] != INF)
                {
                    ans = Math.Min(ans, best[j] + length);
                }

                best[i] = Math.Min(best[i], length);
            }

            // Store latest occurrence
            prefix[sum] = i;
        }

        return ans == INF ? -1 : ans;
    }
}

File: 2026/09-September/17-Thursday/solution.java
import java.util.*;

class Solution {
    public int minSumOfLengths(int[] arr, int target) {
        int n = arr.length;
        int INF = 1_000_000_000;

        // Prefix sum -> latest index
        HashMap<Integer, Integer> map = new HashMap<>();
        map.put(0, 0);

        // best[i] = minimum length of a valid subarray
        // completely inside the first i elements
        int[] best = new int[n + 1];
        Arrays.fill(best, INF);

        int sum = 0;
        int ans = INF;

        for (int i = 1; i <= n; i++) {
            sum += arr[i - 1];

            // Carry forward the previous best
            best[i] = best[i - 1];

            // We need an earlier prefix sum of sum - target
            if (map.containsKey(sum - target)) {
                int j = map.get(sum - target);

                // Subarray [j, i-1] has sum = target
                int len = i - j;

                // best[j] is completely before this subarray,
                // so they do not overlap
                if (best[j] != INF) {
                    ans = Math.min(ans, best[j] + len);
                }

                // Update best for the current prefix
                best[i] = Math.min(best[i], len);
            }

            // Store latest occurrence of this prefix sum
            map.put(sum, i);
        }

        return ans == INF ? -1 : ans;
    }
}
